src/main/java/com/epam/ta/reportportal/core/filter/impl/DeleteUserFilterHandlerImpl.java

```diff
@@ -20,37 +20,46 @@
 import com.epam.ta.reportportal.auth.acl.ReportPortalAclHandler;
 import com.epam.ta.reportportal.core.events.MessageBus;
 import com.epam.ta.reportportal.core.events.activity.FilterDeletedEvent;
-import com.epam.ta.reportportal.core.filter.GetUserFilterHandler;
 import com.epam.ta.reportportal.core.filter.IDeleteUserFilterHandler;
 import com.epam.ta.reportportal.dao.UserFilterRepository;
 import com.epam.ta.reportportal.entity.filter.UserFilter;
+import com.epam.ta.reportportal.exception.ReportPortalException;
 import com.epam.ta.reportportal.ws.model.OperationCompletionRS;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.function.Predicate;
+
+import static com.epam.ta.reportportal.commons.validation.BusinessRule.expect;
 import static com.epam.ta.reportportal.ws.converter.converters.UserFilterConverter.TO_ACTIVITY_RESOURCE;
+import static com.epam.ta.reportportal.ws.model.ErrorType.USER_FILTER_NOT_FOUND;
 
 @Service
 public class DeleteUserFilterHandlerImpl implements IDeleteUserFilterHandler {
 
 	private final UserFilterRepository userFilterRepository;
+
 	private final MessageBus messageBus;
-	private final GetUserFilterHandler getFilterHandler;
 
-	@Autowired
-	private ReportPortalAclHandler aclHandler;
+	private final ReportPortalAclHandler aclHandler;
 
 	@Autowired
 	public DeleteUserFilterHandlerImpl(UserFilterRepository userFilterRepository, MessageBus messageBus,
-			GetUserFilterHandler getFilterHandler) {
+			ReportPortalAclHandler aclHandler) {
 		this.userFilterRepository = userFilterRepository;
 		this.messageBus = messageBus;
-		this.getFilterHandler = getFilterHandler;
+		this.aclHandler = aclHandler;
 	}
 
 	@Override
 	public OperationCompletionRS deleteFilter(Long id, ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user) {
-		UserFilter userFilter = getFilterHandler.getFilter(id, projectDetails, user);
```

Comment:
you can't replace this with just find by id case only user with ADMINISTRATION acl permission could delete filters